packages/backend/src/core/sync/BlockDownloader.ts

```diff
@@ -12,68 +10,77 @@ import { createEventEmitter } from '../../tools/EventEmitter'
 import { JobQueue } from '../../tools/JobQueue'
 import { Logger } from '../../tools/Logger'
 
-export interface KnownBlock {
-  readonly number: number
-  readonly hash: Hash256
+export interface BlockDownloaderEvents {
+  newBlock: BlockRecord
+  reorg: BlockRecord[]
 }
 
-type State =
-  | { type: 'not-started' }
-  | { type: 'working'; lastKnownBlock: KnownBlock }
+const SAFE_BLOCK_DISTANCE = 100
 
 export class BlockDownloader {
   private events = createEventEmitter<BlockDownloaderEvents>()
-  private state: State = { type: 'not-started' }
   private jobQueue: JobQueue
 
+  private lastKnown = 0
+  private queueTip = 0
+  private started = false
+
   constructor(
     private ethereumClient: EthereumClient,
     private blockRepository: BlockRepository,
-    private logger: Logger
+    private logger: Logger,
+    private safeBlockDistance = SAFE_BLOCK_DISTANCE
   ) {
     this.logger = this.logger.for(this)
     this.jobQueue = new JobQueue({ maxConcurrentJobs: 1 }, this.logger)
   }
 
   async start() {
-    const lastKnownBlock = await this.blockRepository.getLast()
+    this.started = true
+    this.lastKnown = (await this.blockRepository.getLast())?.number ?? 0
+    this.queueTip = await this.ethereumClient.getBlockNumber()
```

Comment:
I wouldn't change the name of this property, but...

![image](https://user-images.githubusercontent.com/15332326/153448577-91c5c9b7-9b2f-4aa1-9445-69ed6f8ceee1.png)